autogpt_platform/backend/backend/blocks/google/sheets.py

```diff
@@ -190,3 +261,790 @@ def _write_sheet(
             .execute()
         )
         return result
+
+
+class GoogleSheetsAppendBlock(Block):
+    class Input(BlockSchema):
+        credentials: GoogleCredentialsInput = GoogleCredentialsField(
+            ["https://www.googleapis.com/auth/spreadsheets"]
+        )
+        spreadsheet_id: str = SchemaField(description="Spreadsheet ID")
+        sheet_name: str = SchemaField(
+            description="Optional sheet to append to (defaults to first sheet)",
+            default="",
+        )
+        values: list[list[str]] = SchemaField(description="Rows to append")
+
+    class Output(BlockSchema):
+        result: dict = SchemaField(description="Append API response")
+        error: str = SchemaField(description="Error message, if any")
+
+    def __init__(self):
+        super().__init__(
+            id="531d50c0-d6b9-4cf9-a013-7bf783d313c7",
+            description="Append data to a Google Sheet (sheet optional)",
+            categories={BlockCategory.DATA},
+            input_schema=GoogleSheetsAppendBlock.Input,
+            output_schema=GoogleSheetsAppendBlock.Output,
+            disabled=not GOOGLE_OAUTH_IS_CONFIGURED,
```

Comment:
Disable this in prod too, since OAuth will be configured but we aren’t approved to use sheets in prod